staging/src/k8s.io/api/core/v1/types.go

```diff
@@ -2658,6 +2658,12 @@ const (
 	// disruption (such as preemption, eviction API or garbage-collection).
 	// The constant is to be renamed once the name is accepted within the KEP-3329.
 	AlphaNoCompatGuaranteeDisruptionTarget PodConditionType = "DisruptionTarget"
+	// ResourceExhausted indicates the pod is in the Failed phase or is about to
+	// transition into the Failed phase (and is about to be deleted) due to either:
+	// - exceeding its ephemeral storage limits; or
+	// - running an OOM killed container when the pod's .spec.restartPolicy=Never.
```

Comment:
when it exceeds ephemeral storage limits, is it always applied or only if restartPolicy=Never?

Reply:
In this case the condition is added regardless od restartpolicy as the entire pod is evicted by eviction_manager